### ISSUE TITLE
Unable to load asset *translation path*

### ISSUE THREAD

**Issue by naamapps** [aissat/easy_localization#455]
Hey,
with latest version easy_localization: ^3.0.1-dev.

I get this error when opening the app.
`Unable to load asset *translation path*`

Please help,
Thanks

**Comment by kareemzok** [comment]
I got the same with the version 3.0.0

**Comment by kareemzok** [comment]
were you able to resolve it?

**Comment by naamapps** [comment]
Nope.. I think it's a bug with the new release.
I don't get the error in version 3.0.0

**Comment by kareemzok** [comment]
Hello,
I was able to resolve it - it was missing the data in ios/Runner/Info.plist

similar to below

```
<key>CFBundleLocalizations</key>
<array>
	<string>en</string>
	<string>nb</string>
</array>
```


check https://pub.dev/packages/easy_localization ⚠️ Note on iOS [#](https://pub.dev/packages/easy_localization#-note-on-ios)

**Comment by naamapps** [comment]
I was getting the error on Android 😅

**Comment by kareemzok** [comment]
Yea  same for me but i was missing the code i mentioned when i implemented i didnt add ios part - if yo u still get it then make sure you have assets path correct, it should be on same root of lib 

![image](https://user-images.githubusercontent.com/6444765/153920550-f496dc80-0ddf-4bc9-9a4f-9331c2428f7c.png)

**Comment by naamapps** [comment]
It works with version 3.0.0 so I guess everything is configured correctly.
Only with the new version it fails.. I don't think the issue is on my part

**Comment by kareemzok** [comment]
Ah okay then maybe cause your using 3.0.1 dev 

Its not preferable to use a dev version as it may contains lot of issues

**Comment by kareemzok** [comment]
Ah okay then maybe cause your using 3.0.1 dev 

Its not preferable to use a dev version as it may contains lot of issues

**Comment by naamapps** [comment]
This is the issue.
I don't have a translation with only the language code.
I only have translations with language + country (en-US for example).
 
This `if` statement doesn't make sense.
@aissat @Overman775 @shrdnar
https://github.com/aissat/easy_localization/blob/992f9804d9b0bde034c82c3ca521d3e7a81488ea/lib/src/easy_localization_controller.dart#L84-L86

**Comment by kareemzok** [comment]
Ahh okay - i used it with the language code only and it works. 
but in the function you seam passing only code to the locale. Maybe you should pass both lang and code

**Comment by aissat** [comment]
Hi @naamapps,  please if u can give us more logs and when u tested  app "OS" ?

And u can try this `startLocale`  like here:

```dart
EasyLocalization(
        supportedLocales: const [Locale('ar', 'DZ'), Locale('en', 'US')],
        startLocale: const Locale('ar', 'DZ'),
        path: 'assets/locales',
        child: const App(),
      ),
```

**Comment by naamapps** [comment]
Hello @aissat,
There are no logs I can give you as this error is the only thing that is shown in the console.

I should mention that the error disappeared when setting `useFallbackTranslations` to false.

Setting the start locale does nothing.

I think the problem is in the code I commented in my last message here. The if statement doesn't make sense to me. It checks if the country code is valid and then uses the language code (which doesn't exist in my project).

**Comment by aissat** [comment]
> Hello @aissat, There are no logs I can give you as this error is the only thing that is shown in the console.
> 
> I should mention that the error disappeared when setting `useFallbackTranslations` to false.
> 
> Setting the start locale does nothing.
> 
> I think the problem is in the code I commented in my last message here. The if statement doesn't make sense to me. It checks if the country code is valid and then uses the language code (which doesn't exist in my project).

@naamapps  I checked  everything (code, tests) working well.
**What's the target OS ??!!**

**Comment by naamapps** [comment]
The os is android 12.

**Comment by naamapps** [comment]
And again mentioning this happens only on latest version - ^3.0.1-dev

**Comment by aissat** [comment]
> And again mentioning this happens only on latest version - ^3.0.1-dev

I'll check it, thanks .

**Comment by naamapps** [comment]
Hey @aissat, any updates?

**Comment by Slh-Ibrahim** [comment]
Declare your assets localization directory in pubspec.yaml:

flutter:
  assets:
    - assets/translations/

**Comment by naamapps** [comment]
@DurraDev1 I did that.
As I mentioned before this happens only on latest version - ^3.0.1-dev.

**Comment by shrdnar** [comment]
> This is the issue. I don't have a translation with only the language code. I only have translations with language + country (en-US for example).
> 
> This `if` statement doesn't make sense. @aissat @Overman775 @shrdnar
> 
> https://github.com/aissat/easy_localization/blob/992f9804d9b0bde034c82c3ca521d3e7a81488ea/lib/src/easy_localization_controller.dart#L84-L86

I've submitted a PR (https://github.com/aissat/easy_localization/pull/470) to prevent `loadTranslations` from stopping/crashing if the base language asset is not found.

The if statement does make sense when `useFallbackTranslations` is true. The idea here is that the majority of your translation keys and values can exist in a base language asset file (es.json for example) and allows you to override values for specific keys for a particular region or country (es-MX.json) without having to specify or duplicate all the other keys and values that would otherwise remain unchanged. It simplifies maintainability if you deal with a lot of languages or a lot of keys/values and need to make a change. You can refer to the original PR ([#440](https://github.com/aissat/easy_localization/pull/470)) for more info.

**Comment by naamapps** [comment]
Thanks for the PR and the explanation.

**Comment by aea91** [comment]
Version is 3.0, I face that problem on only ios. I checked the settings all correct. My app is rejected on App Store because of that problem cause I didn't face it on ios simulator.

**Comment by naamapps** [comment]
Closing because for me this is fixed in 3.0.1.
Thanks all
